Tests/SonarScanner.MSBuild.PreProcessor.Test/AnalysisConfigGeneratorTests.cs

```diff
@@ -384,6 +384,47 @@ public void GenerateFile_ExcludeCoverage_Exclusions_MultipleSpecified()
             .Which.Value.Should().Be("coverage1.xml,coverage2.xml,coverage3.xml,coverage4.xml");
     }
 
+    [DataTestMethod]
+    [DataRow("coverage.xml", "", "", "", "coverage.xml", "")]
+    [DataRow("coverage.xml", "", "local.cs,local.js", "", "local.cs,local.js,coverage.xml", "")]
+    [DataRow("coverage.xml", "", "", "server.cs,server.js", "", "server.cs,server.js,coverage.xml")]
+    [DataRow("coverage.xml", "", "local.cs,local.js", "server.cs,server.js", "local.cs,local.js,coverage.xml", "server.cs,server.js")]
+    [DataRow("", "", "", "", "", "")]
+    [DataRow("", "", "local.cs,local.js", "", "local.cs,local.js", "")]
+    [DataRow("", "", "", "server.cs,server.js", "", "server.cs,server.js")]
+    [DataRow("", "", "local.cs,local.js", "server.cs,server.js", "local.cs,local.js", "server.cs,server.js")]
+    [DataRow("", "coverage.xml", "", "", "coverage.xml", "")]
+    [DataRow("", "coverage.xml", "local.cs,local.js", "", "local.cs,local.js,coverage.xml", "")]
+    [DataRow("", "coverage.xml", "", "server.cs,server.js", "", "server.cs,server.js,coverage.xml")]
+    [DataRow("", "coverage.xml", "local.cs,local.js", "server.cs,server.js", "local.cs,local.js,coverage.xml", "server.cs,server.js")]
+    [DataRow("localCoverage.xml", "serverCoverage.xml", "", "", "localCoverage.xml", "")]
+    [DataRow("localCoverage.xml", "serverCoverage.xml", "local.cs,local.js", "", "local.cs,local.js,localCoverage.xml", "")]
+    [DataRow("localCoverage.xml", "serverCoverage.xml", "", "server.cs,server.js", "", "server.cs,server.js,localCoverage.xml")]
```

Comment:
I am a bit unsure about this case.
Did you test what happens if local exclusions are empty?

`sonar-project.properties` will look like this:

```xml
sonar.exclusions= <SERVER_VALUES>
blabla...
sonar.exclusions=
// local, nothing here^, but this might override the previous one
```

I think ideally you would want the local exclusions to not even exist as a property in the collection.

Reply:
You cannot provide an empty `/d:sonar.exclusions=""` parameter from the command line, it's invalid 👍

Reply:
I'll add some tests where the property does not exist at all, as you pointed our [here](https://github.com/SonarSource/sonar-scanner-msbuild/pull/2184#discussion_r1753560919)